### DISCUSSION TITLE
Get Mouse position in the canva

### DISCUSSION THREAD

**Discussion by AFinne-Specks** [xyflow/xyflow#3112] [Q&A] [answered]
Hello, 
I would like to get the position on the mouse in the canva. I got the mouse position in the dom element but how could i get the actual position in the canva.
For example i would like to create a node where i click.

Thanks !

**Comment by bcakmakoglu** [comment, marked as answer]
[You can use the `project` function provided by the react flow instance.](https://reactflow.dev/docs/api/react-flow-instance/#project)

```ts
const position = reactFlowInstance.project({
  x: event.clientX - reactFlowBounds.left,
  y: event.clientY - reactFlowBounds.top
});
```

[Here's an example.](https://codesandbox.io/s/flamboyant-montalcini-d53y55?file=/src/Flow.tsx:1661-1817)

**Reply by AFinne-Specks** [reply]
Thanks a lot !
